distribution/lib/Standard/Visualization/0.0.0-dev/src/Widgets.enso

```diff
@@ -1,15 +1,26 @@
 from Standard.Base import all
+from Standard.Base.Meta import Instrumentor
+import Standard.Base.Errors.Common.Not_Invokable
 
 ## PRIVATE
    Basic preprocessor for widgets metadata visualization.
 
    Returns full annotation data for all requested arguments.
-get_widget_json : Any -> Any -> Vector Text -> Text
-get_widget_json value call_name argument_names =
+get_widget_json : Any -> Any -> Vector Text -> Text -> Text
+get_widget_json value call_name argument_names uuids="{}" =
+    other_args = uuids.parse_json
+
+    cache (name_or_index:Text|Integer) -> Any =
+        arg = other_args.get name_or_index.to_text
+        case arg of
+            uuid:Text -> Instrumentor.uuid uuid
+            _ -> Nothing
+
     read_annotation argument =
         annotation = Warning.clear <| Meta.get_annotation value call_name argument
-        case annotation of
-            _ : Function -> annotation value
-            _ -> annotation
+        return_target err = err.payload.target
+        Panic.catch Not_Invokable handler=return_target
+            annotation value cache=cache
```

Comment:
The `cache` argument is _"optional"_. If the function doesn't accept it, `Not_Invokable` panic is going to be raised. It is caught and the value that refused to accept `cache` argument is returned instead. As a result it is **not necessary** to accept the `cache` argument in dynamic dropdown functions.

Reply:
that is very subtle!

Reply:
Subtle, but tested!